modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1362,6 +1364,26 @@ private void _validateName(
 		}
 	}
 
+	private void _validateObjectDefinitionStorageType(
+			ObjectDefinition objectDefinition, String businessType)
+		throws PortalException {
+
+		if (objectDefinition.getStorageType(
+			).equals(
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE
+			) &&
+			(businessType.equals(
+				ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION) ||
+			 businessType.equals(
+				 ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT))) {
+
+			throw new ObjectFieldBusinessTypeException(
```

Comment:
"Aggregation and Attachment business types are not supported for the Salesforce storage type"